clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -711,12 +711,15 @@ private void processDisconnection(List<ClientResponse> responses,
                 AuthenticationException exception = disconnectState.exception();
                 connectionStates.authenticationFailed(nodeId, now, exception);
                 metadataUpdater.handleAuthenticationFailure(exception);
-                log.error("Connection to node {} ({}) failed authentication due to: {}", nodeId, disconnectState.remoteAddress(), exception.getMessage());
+                log.error("Connection to node {} ({}) failed authentication due to: {}", nodeId,
+                    disconnectState.remoteAddress(), exception.getMessage());
                 break;
             case AUTHENTICATE:
-                // This warning applies to older brokers which don't provide feedback on authentication failures
-                log.warn("Connection to node {} ({}) terminated during authentication. This may indicate " +
-                        "that authentication failed due to invalid credentials.", nodeId, disconnectState.remoteAddress());
+                log.warn("Connection to node {} ({}) terminated during authentication. This may happen " +
+                    "due to any of the following reasons: (1) Authentication failed due to invalid " +
+                    "credentials with brokers older than 1.0.0, (2) Firewall blocking Kafka TLS " +
+                    "traffic (eg it may only allow HTTPS traffic), (3) Transient network issue.",
+                    nodeId, disconnectState.remoteAddress());
```

Comment:
@rajinisivaram Is this message accurate?

Reply:
yes sounds good.